### ISSUE TITLE
Bug report

### ISSUE THREAD

**Issue by gonejack** [ermadan/zoolytic#1]
Hi author, I found a bug on recent intelliJ.

```
IntelliJ IDEA 2019.3 (Ultimate Edition)
Build #IU-193.5233.102, built on November 28, 2019
Subscription is active until April 18, 2020
Runtime version: 11.0.4+10-b520.11 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.15.1
GC: ParNew, ConcurrentMarkSweep
Memory: 1981M
Cores: 8
Registry: debugger.watches.in.variables=false, ide.tooltip.initialDelay=0, js.debugger.webconsole=false
Non-Bundled Plugins: AceJump, com.intellij.plugins.watcher, CMD Support, FrameSwitcher, GrepConsole, Pythonid, com.alayouni.ansiHighlight, com.intellij.ideolog, com.jetbrains.jbre.bintray.downloader, com.jetbrains.lang.ejs, com.jetbrains.php, com.jetbrains.plugins.ini4idea, com.jetbrains.plugins.jade, com.vladsch.idea.multimarkdown, de.cmm.side_nodes, de.netnexus.camelcaseplugin, io.protostuff.protostuff-jetbrains-plugin, mobi.hsz.idea.gitignore, org.jetbrains.kotlin, org.jetbrains.plugins.go-template, org.jetbrains.plugins.go, org.jetbrains.plugins.hocon, org.toml.lang, org.zoolytic, ru.adelf.idea.dotenv, some.awesome
```

```
java.lang.ClassCastException: class org.zoolytic.plugin.ZkTreeNode cannot be cast to class org.zoolytic.plugin.ZkRootTreeNode (org.zoolytic.plugin.ZkTreeNode and org.zoolytic.plugin.ZkRootTreeNode are in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @23594745)
	at org.zoolytic.plugin.ZoolyticToolWindowFactory.removeCluster(ZoolyticToolWindowFactory.kt:285)
	at org.zoolytic.plugin.ZoolyticToolWindowFactory.access$removeCluster(ZoolyticToolWindowFactory.kt:45)
	at org.zoolytic.plugin.ZoolyticToolWindowFactory$RemoveAction.actionPerformed(ZoolyticToolWindowFactory.kt:372)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:266)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:283)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAwareWithCallbacks(ActionUtil.java:257)
	at com.intellij.openapi.actionSystem.impl.ActionButton.actionPerformed(ActionButton.java:194)
	at com.intellij.openapi.actionSystem.impl.ActionButton.performAction(ActionButton.java:157)
	at com.intellij.openapi.actionSystem.impl.ActionButton.processMouseEvent(ActionButton.java:447)
	at java.desktop/java.awt.Component.processEvent(Component.java:6416)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5026)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4858)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2773)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4858)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:906)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:844)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:776)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:422)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:698)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:421)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)```

**Comment by ermadan** [comment]
fixed in version  0.5

**Comment by azolotko** [comment]
I'm still observing this when trying to remove a node in version 0.5.1:

```
java.lang.ClassCastException: class org.zoolytic.plugin.ZkTreeNode cannot be cast to class org.zoolytic.plugin.ZkRootTreeNode (org.zoolytic.plugin.ZkTreeNode and org.zoolytic.plugin.ZkRootTreeNode are in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @3d21ca48)
	at org.zoolytic.plugin.MainWindow.removeCluster(MainWindow.kt:265)
	at org.zoolytic.plugin.MainWindow.access$removeCluster(MainWindow.kt:42)
	at org.zoolytic.plugin.MainWindow$RemoveAction.actionPerformed(MainWindow.kt:323)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:298)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:315)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAwareWithCallbacks(ActionUtil.java:289)
	at com.intellij.openapi.actionSystem.impl.ActionButton.actionPerformed(ActionButton.java:194)
	at com.intellij.openapi.actionSystem.impl.ActionButton.performAction(ActionButton.java:157)
	at com.intellij.openapi.actionSystem.impl.ActionButton.processMouseEvent(ActionButton.java:447)
	at java.desktop/java.awt.Component.processEvent(Component.java:6416)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5026)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4858)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2773)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4858)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:908)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:846)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:778)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:424)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:698)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:423)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

```
IntelliJ IDEA 2019.3.2 Preview (Community Edition)
Build #IC-193.6015.11, built on December 27, 2019
Runtime version: 11.0.5+10-b520.30 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.15.2
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 8
Registry: ide.balloon.shadow.size=0
Non-Bundled Plugins: DBN, Docker, FrameSwitcher, Jetbrains TeamCity Plugin, Key Promoter X, PythonCore, VisualVMLauncher, com.atlassian.bitbucket.references, com.chrisrm.idea.MaterialThemeUI, io.protostuff.protostuff-jetbrains-plugin, izhangzhihao.rainbow.brackets, org.intellij.scala, org.jetbrains.plugins.hocon, org.toml.lang, org.rust.lang, org.scalafmt.ScalaFmt, org.zoolytic, training
```
